src/containerapp/azext_containerapp/_clients.py

```diff
@@ -1280,6 +1286,16 @@ def list_files(cls, cmd, identifier, path, session_pool_endpoint):
 
         r = send_raw_request(cmd.cli_ctx, "GET", request_url, resource=SESSION_RESOURCE)
         return r.json()
+    
+    @classmethod
+    def extract_path_from_filename(cls, path, filename):
+        if '/' not in filename:
+            return path, filename
+        path_in_filename, filename = filename.rsplit('/', 1)
+        if path is None:
+            return path_in_filename, filename
+        else:
+            return path.rstrip("/") + "/" + path_in_filename.lstrip('/'), filename
```

Comment:
this should be a static method.

```suggestion
    @staticmethod
    def extract_path_from_filename(path, filename):
        if '/' not in filename:
            return path, filename
        path_in_filename, filename = filename.rsplit('/', 1)
        if path is None:
            return path_in_filename, filename

        full_path = f"{path.rstrip('/')}/{path_in_filename.lstrip('/')}"
        return full_path, filename
```